portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,24 +25,61 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+     ArrayList<String> StringList = new ArrayList<String>();
 
 
+ public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "thisName", "");
+    text += "\n";
+    text +=   getParameter(request, "thisComments", "");
+    boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+    boolean sort = Boolean.parseBoolean(getParameter(request, "sort", "false"));
 
-  @Override
+    // Convert the text to upper case.
+    if (upperCase) {
+      text = text.toUpperCase();
+    }
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
+
+    // Sort the words.
+    if (sort) {
+      Arrays.sort(words);
+    }
+
+    StringList.add(Arrays.toString(words));
+
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println(Arrays.toString(words));
+
+   response.sendRedirect("/contacts.html");
```

Comment:
You can't write a response and also redirect - which one were you planning to do?

Reply:
Ah I tested it and it works - which surprises me, but it's true. Printing the words here will have no effect, but OK, you can fix this later.